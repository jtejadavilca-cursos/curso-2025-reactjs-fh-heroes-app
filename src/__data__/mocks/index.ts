export { mockHero } from "./mock-hero";
export { mockSummaryData } from "./mock-summary-data";
