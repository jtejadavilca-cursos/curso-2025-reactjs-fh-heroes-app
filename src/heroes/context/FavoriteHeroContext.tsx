import { createContext, useEffect, useState, type PropsWithChildren } from "react";
import type { Hero } from "../types/hero.interface";

interface FavoriteHeroContextProps {
    // State
    favorites: Hero[];
    favoritesCount: number;

    // Methods
    isFavorite: (hero: Hero) => boolean;
    toggleFavorite: (hero: Hero) => void;
    calculatePercentage: (total: number) => string;
}

export const FavoriteHeroContext = createContext<FavoriteHeroContextProps>({} as FavoriteHeroContextProps);

const FAVORITES_LOCAL_STORAGE_KEY = "favorites";

const getFavoritesfromLocalStorage = (): Hero[] => {
    const favorites = localStorage.getItem(FAVORITES_LOCAL_STORAGE_KEY);
    return favorites ? JSON.parse(favorites) : [];
};

export const FavoriteHeroProvider = ({ children }: PropsWithChildren) => {
    const [favorites, setFavorites] = useState<Hero[]>(getFavoritesfromLocalStorage());

    const isFavorite = (hero: Hero) => {
        return favorites.some((h) => h.id === hero.id);
    };

    const toggleFavorite = (hero: Hero) => {
        const heroExists = favorites.find((h) => h.id === hero.id);

        if (heroExists) {
            const newFavorites = favorites.filter((h) => h.id !== hero.id);
            setFavorites(newFavorites);
            return;
        }
        setFavorites([...favorites, hero]);
    };

    const calculatePercentage = (total: number): string => {
        if (total <= 0) return "0.00";

        return ((favorites.length / total) * 100).toFixed(2);
    };

    useEffect(() => {
        localStorage.setItem(FAVORITES_LOCAL_STORAGE_KEY, JSON.stringify(favorites));
    }, [favorites]);

    return (
        <FavoriteHeroContext
            value={{
                favoritesCount: favorites.length,
                favorites,
                isFavorite,
                toggleFavorite,
                calculatePercentage,
            }}
        >
            {children}
        </FavoriteHeroContext>
    );
};
