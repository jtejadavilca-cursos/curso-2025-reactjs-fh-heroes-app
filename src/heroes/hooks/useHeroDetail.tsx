import { useQuery } from "@tanstack/react-query";
import { getHeroeByIdOsSlugAction } from "../actions/get-hero.action";

export const useHeroDetail = (idSlug: string) => {
    const { data: superheroData } = useQuery({
        queryKey: ["heroes", "idSlug", idSlug],
        queryFn: () => getHeroeByIdOsSlugAction(idSlug),
        staleTime: 1000 * 60 * 5, // 5 minutos
    });

    const totalPower = superheroData
        ? superheroData.strength + superheroData.intelligence + superheroData.speed + superheroData.durability
        : 0;
    const averagePower = Math.round((totalPower / 4) * 10);

    return {
        superheroData,
        totalPower,
        averagePower,
    };
};
