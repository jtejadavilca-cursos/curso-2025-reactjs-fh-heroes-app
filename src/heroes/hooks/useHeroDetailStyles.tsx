// const mapStatusColors = new Map<HeroStatus, string>();
const mapStatusColors = new Map<string, string>();
mapStatusColors.set("activo", "bg-green-500");
mapStatusColors.set("inactivo", "bg-gray-500");
mapStatusColors.set("retirado", "bg-blue-500");

// const mapCategoryColors = new Map<HeroCategory, string>();
const mapCategoryColors = new Map<string, string>();
mapCategoryColors.set("héroe", "bg-blue-500");
mapCategoryColors.set("villano", "bg-red-500");
mapCategoryColors.set("antihéroe", "bg-purple-500");

const getStatusColor = (status: string) => mapStatusColors.get(status.toLowerCase()) ?? "bg-gray-500";
const getCategoryColor = (category: string) => mapCategoryColors.get(category.toLowerCase()) ?? "bg-gray-500";

export const useHeroDetailStyles = () => {
    return {
        getStatusColor,
        getCategoryColor,
    };
};
