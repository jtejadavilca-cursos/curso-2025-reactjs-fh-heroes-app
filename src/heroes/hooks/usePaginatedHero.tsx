import { useQuery } from "@tanstack/react-query";
import { getHeroesByPageAction } from "../actions/get-heroes-by-page.action";

export const usePaginatedHero = (page: number, limit: number, category: string = "all") => {
    const sanitizedPage = isNaN(page) ? 1 : +page;
    const sanitizedLimit = isNaN(limit) ? 6 : +limit;

    return useQuery({
        queryKey: ["heroes", { page: sanitizedPage, limit: sanitizedLimit, category }],
        queryFn: () => getHeroesByPageAction(sanitizedPage, sanitizedLimit, category),
        staleTime: 1000 * 60 * 5, // 5 minutos
    });
};
