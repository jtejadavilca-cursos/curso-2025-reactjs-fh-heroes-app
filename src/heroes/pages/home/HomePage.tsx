import { use } from "react";
import { Heart } from "lucide-react";

import { Tabs, TabsContent, TabsList, TabsTrigger } from "@/components/ui/tabs";
import { CustomJumbotron } from "@/components/custom/CustomJumbotron";
import { HeroStats } from "@/heroes/components/HeroStats";
import { HeroGrid } from "@/heroes/components/HeroGrid";
import { CustomPagination } from "@/components/custom/CustomPagination";
import { useSearchParams } from "react-router";
import { useHeroSummary } from "@/heroes/hooks/useHeroSummary";
import { usePaginatedHero } from "@/heroes/hooks/usePaginatedHero";
import { FavoriteHeroContext } from "@/heroes/context/FavoriteHeroContext";

type TabsType = "all" | "favorites" | "hero" | "villain";

export const HomePage = () => {
    const [searchParams, setSearchParams] = useSearchParams();

    const category = searchParams.get("tab") ?? "all";
    const page = searchParams.get("page") ?? 1;
    const limit = searchParams.get("limit") ?? 6;

    const { data: heroesResponse } = usePaginatedHero(+page, +limit, category);
    const { data: summary } = useHeroSummary();
    const { favoritesCount, favoritesPaginated, totalFavoritePages } = use(FavoriteHeroContext);

    const handleClickTab = (category: TabsType) => {
        setSearchParams((prev) => {
            prev.set("tab", category);
            prev.set("page", "1");
            return prev;
        });
    };

    return (
        <>
            {/* Header */}
            <CustomJumbotron
                title="Universo de Superheroes!"
                description="Descubre, explora, y administra tus superhéroes y villanos favoritos!"
            />

            {/* Stats Dashboard */}
            <HeroStats />

            {/* Tabs */}
            <Tabs value={category} className="mb-8">
                <TabsList className="grid w-full grid-cols-4">
                    <TabsTrigger
                        onClick={() => handleClickTab("all")}
                        value="all"
                        data-testid="tab-trigger-summary-all-heroes"
                    >
                        All Characters ({summary?.totalHeroes})
                    </TabsTrigger>
                    <TabsTrigger
                        onClick={() => handleClickTab("favorites")}
                        value="favorites"
                        className="flex items-center gap-2"
                        data-testid="tab-trigger-summary-favorites-heroes"
                    >
                        <Heart className="h-4 w-4" />
                        Favorites ({favoritesCount})
                    </TabsTrigger>
                    <TabsTrigger
                        onClick={() => handleClickTab("hero")}
                        value="hero"
                        data-testid="tab-trigger-summary-only-heroes"
                    >
                        Heroes ({summary?.heroCount})
                    </TabsTrigger>
                    <TabsTrigger
                        onClick={() => handleClickTab("villain")}
                        value="villain"
                        data-testid="tab-trigger-summary-only-villain"
                    >
                        Villains ({summary?.villainCount})
                    </TabsTrigger>
                </TabsList>

                <TabsContent value="all" data-testid="tab-content-summary-all-heroes">
                    <h1>Todos los personales</h1>
                </TabsContent>
                <TabsContent value="favorites" data-testid="tab-content-summary-favorites-heroes">
                    <h1>Favoritos</h1>
                </TabsContent>
                <TabsContent value="hero" data-testid="tab-content-summary-only-heroes">
                    <h1>Héroes!</h1>
                </TabsContent>
                <TabsContent value="villain" data-testid="tab-content-summary-only-villain">
                    <h1>Villanos</h1>
                </TabsContent>
            </Tabs>

            {/* Character Grid */}
            {category === "favorites" ? (
                <HeroGrid heroes={favoritesPaginated} />
            ) : (
                <HeroGrid heroes={heroesResponse?.heroes ?? []} />
            )}

            {/* Pagination */}
            {category === "favorites" ? (
                <CustomPagination totalPages={totalFavoritePages} />
            ) : (
                <CustomPagination totalPages={heroesResponse?.pages ?? 0} />
            )}
        </>
    );
};

export default HomePage;
