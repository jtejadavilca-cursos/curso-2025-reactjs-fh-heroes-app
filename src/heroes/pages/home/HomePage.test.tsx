import { fireEvent, render, screen } from "@testing-library/react";
import { beforeEach, describe, expect, test, vi } from "vitest";
import HomePage from "./HomePage";
import { MemoryRouter } from "react-router";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { usePaginatedHero } from "@/heroes/hooks/usePaginatedHero";
import { mockHero } from "@/__data__/mocks";
import type { Hero } from "@/heroes/types/hero.interface";
import { useHeroSummary } from "@/heroes/hooks/useHeroSummary";
import type { SummaryInformationResponse } from "@/heroes/types/summary-information.response";
import { FavoriteHeroProvider } from "@/heroes/context/FavoriteHeroContext";
import { mockHeros } from "@/__data__/mocks/mock-hero";

// Mocked pages
const mockJumbotronContent = "Mocked CustomJumbotron";
const mockHeroStatsContent = "Mocked HeroStats section";
const mockHeroGridContent = "Mocked Hero Grid";
vi.mock("@/components/custom/CustomJumbotron", () => ({
    CustomJumbotron: () => <div data-testid="mocked-jumbotron-header">{mockJumbotronContent}</div>,
}));
vi.mock("@/heroes/components/HeroStats", () => ({
    HeroStats: () => <div data-testid="mocked-herostats-section">{mockHeroStatsContent}</div>,
}));

vi.mock("@/heroes/components/HeroGrid", () => ({
    HeroGrid: ({ heroes }: { heroes: Hero[] }) => (
        <div data-testid="mocked-hero-grid-section">
            <h3>{mockHeroGridContent}</h3>
            {heroes.map((h) => (
                <div key={h.id} data-testid={`card-${h.id}`}>
                    <h2>Hero {h.name}</h2>
                </div>
            ))}
        </div>
    ),
}));

vi.mock("@/components/custom/CustomPagination", () => ({
    CustomPagination: ({ totalPages }: { totalPages: number }) => (
        <div data-testid="mocked-paginator-section">
            <h3>Mocked CustomPagination section</h3>
            <h2>Total pages : {totalPages}</h2>
        </div>
    ),
}));
// End Mocked pages

// Mocked hooks
//usePaginatedHero
vi.mock("@/heroes/hooks/usePaginatedHero");
const mockUsePaginatedHero = vi.mocked(usePaginatedHero);

vi.mock("@/heroes/hooks/useHeroSummary");
const mockUseHeroSummary = vi.mocked(useHeroSummary);
// End Mocked hooks

const queryClient = new QueryClient({
    defaultOptions: {
        queries: {
            retry: false,
        },
    },
});

const renderHomePage = (
    initialEntries?: string[], //
    summary?: Partial<SummaryInformationResponse>,
    heroes: Hero[] = [mockHero], //
    pages: number = 1 //
) => {
    mockUsePaginatedHero.mockReturnValue({
        data: {
            heroes,
            pages,
        },
    } as unknown as ReturnType<typeof usePaginatedHero>);

    mockUseHeroSummary.mockReturnValue({
        data: summary,
    } as unknown as ReturnType<typeof useHeroSummary>);

    return render(
        <MemoryRouter initialEntries={initialEntries}>
            <FavoriteHeroProvider>
                <QueryClientProvider client={queryClient}>
                    <HomePage />
                </QueryClientProvider>
            </FavoriteHeroProvider>
        </MemoryRouter>
    );
};

describe("Testing HomePage.test", () => {
    beforeEach(() => {
        vi.clearAllMocks();
    });

    test("should render HomePage with default values", () => {
        // With one favorite in localStorage by default
        localStorage.setItem("favorites", JSON.stringify([mockHero]));
        const { container } = renderHomePage([""], {
            totalHeroes: 10,
            heroCount: 6,
            villainCount: 4,
        });

        // Sections.
        expect(screen.getByTestId("mocked-jumbotron-header")).toBeDefined();
        expect(screen.getByTestId("mocked-jumbotron-header").innerHTML).toContain(mockJumbotronContent);
        expect(screen.getByTestId("mocked-herostats-section")).toBeDefined();
        expect(screen.getByTestId("mocked-herostats-section").innerHTML).toContain(mockHeroStatsContent);
        expect(screen.getByTestId("mocked-hero-grid-section")).toBeDefined();
        expect(screen.getByTestId("mocked-hero-grid-section").innerHTML).toContain(mockHeroGridContent);

        // Tabs
        expect(screen.getByTestId("tab-trigger-summary-all-heroes")).toBeDefined();
        expect(screen.getByTestId("tab-trigger-summary-all-heroes").innerHTML).toContain("All Characters (10)");
        expect(screen.getByTestId("tab-trigger-summary-favorites-heroes")).toBeDefined();
        expect(screen.getByTestId("tab-trigger-summary-favorites-heroes").innerHTML).toContain("Favorites (1)");
        expect(screen.getByTestId("tab-trigger-summary-only-heroes")).toBeDefined();
        expect(screen.getByTestId("tab-trigger-summary-only-heroes").innerHTML).toContain("Heroes (6)");
        expect(screen.getByTestId("tab-trigger-summary-only-villain")).toBeDefined();
        expect(screen.getByTestId("tab-trigger-summary-only-villain").innerHTML).toContain("Villains (4)");

        // Tabs content
        expect(screen.getByTestId("tab-content-summary-all-heroes")).toBeDefined();
        expect(screen.getByTestId("tab-content-summary-all-heroes").innerHTML).toContain("Todos los personales");
        expect(screen.getByTestId("tab-content-summary-favorites-heroes")).toBeDefined();
        expect(screen.getByTestId("tab-content-summary-favorites-heroes").innerHTML).toContain("");
        expect(screen.getByTestId("tab-content-summary-only-heroes")).toBeDefined();
        expect(screen.getByTestId("tab-content-summary-only-heroes").innerHTML).toContain("");
        expect(screen.getByTestId("tab-content-summary-only-villain")).toBeDefined();
        expect(screen.getByTestId("tab-content-summary-only-villain").innerHTML).toContain("");

        expect(container).toMatchSnapshot();

        //screen.debug();
    });

    test("should show favorite tab", () => {
        localStorage.setItem("favorites", JSON.stringify(mockHeros));
        const { container } = renderHomePage(["/?page=2&limit=2&tab=favorites"], {
            totalHeroes: 10,
            heroCount: 6,
            villainCount: 4,
        });

        // Sections.
        expect(screen.getByTestId("mocked-jumbotron-header")).toBeDefined();
        expect(screen.getByTestId("mocked-jumbotron-header").innerHTML).toContain(mockJumbotronContent);
        expect(screen.getByTestId("mocked-herostats-section")).toBeDefined();
        expect(screen.getByTestId("mocked-herostats-section").innerHTML).toContain(mockHeroStatsContent);
        expect(screen.getByTestId("mocked-hero-grid-section")).toBeDefined();
        expect(screen.getByTestId("mocked-hero-grid-section").innerHTML).toContain(mockHeroGridContent);

        // Tabs
        expect(screen.getByTestId("tab-trigger-summary-all-heroes")).toBeDefined();
        expect(screen.getByTestId("tab-trigger-summary-all-heroes").innerHTML).toContain("All Characters (10)");
        expect(screen.getByTestId("tab-trigger-summary-favorites-heroes")).toBeDefined();
        expect(screen.getByTestId("tab-trigger-summary-favorites-heroes").innerHTML).toContain("Favorites (5)");
        expect(screen.getByTestId("tab-trigger-summary-only-heroes")).toBeDefined();
        expect(screen.getByTestId("tab-trigger-summary-only-heroes").innerHTML).toContain("Heroes (6)");
        expect(screen.getByTestId("tab-trigger-summary-only-villain")).toBeDefined();
        expect(screen.getByTestId("tab-trigger-summary-only-villain").innerHTML).toContain("Villains (4)");

        // Tabs content
        expect(screen.getByTestId("tab-content-summary-all-heroes")).toBeDefined();
        expect(screen.getByTestId("tab-content-summary-all-heroes").innerHTML).toContain("");
        expect(screen.getByTestId("tab-content-summary-favorites-heroes")).toBeDefined();
        expect(screen.getByTestId("tab-content-summary-favorites-heroes").innerHTML).toContain("Favoritos");
        expect(screen.getByTestId("tab-content-summary-only-heroes")).toBeDefined();
        expect(screen.getByTestId("tab-content-summary-only-heroes").innerHTML).toContain("");
        expect(screen.getByTestId("tab-content-summary-only-villain")).toBeDefined();
        expect(screen.getByTestId("tab-content-summary-only-villain").innerHTML).toContain("");

        // Grid
        expect(screen.getByTestId("mocked-hero-grid-section").children.length).toBe(3);
        expect(screen.getByTestId("mocked-hero-grid-section").innerHTML).toContain(mockHeroGridContent);
        expect(screen.getByTestId("mocked-hero-grid-section").innerHTML).toContain("My Test hero 456");
        expect(screen.getByTestId("mocked-hero-grid-section").innerHTML).toContain("My Test hero 678");

        expect(container).toMatchSnapshot();

        //screen.debug();
    });

    test("should render Villains container when click in villain tab", () => {
        // With one favorite in localStorage by default
        renderHomePage([""], {
            totalHeroes: 10,
            heroCount: 6,
            villainCount: 4,
        });

        // const [allTabd, favoriteTabs, heroesTab, villainsTab] = screen.getAllByRole("tab");
        const [, , , villainsTab] = screen.getAllByRole("tab");
        fireEvent.click(villainsTab);

        expect(mockUsePaginatedHero).toHaveBeenCalledWith(1, 6, "villain");
        //screen.debug();
    });
});
