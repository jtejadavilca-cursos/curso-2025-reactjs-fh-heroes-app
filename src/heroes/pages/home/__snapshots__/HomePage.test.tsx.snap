// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing HomePage.test > should render HomePage with default values 1`] = `
<div>
  <div
    data-testid="mocked-jumbotron-header"
  >
    Mocked CustomJumbotron
  </div>
  <div
    data-testid="mocked-herostats-section"
  >
    Mocked HeroStats section
  </div>
  <div
    class="flex flex-col gap-2 mb-8"
    data-orientation="horizontal"
    data-slot="tabs"
    dir="ltr"
  >
    <div
      aria-orientation="horizontal"
      class="bg-muted text-muted-foreground h-9 items-center justify-center rounded-lg p-[3px] grid w-full grid-cols-4"
      data-orientation="horizontal"
      data-slot="tabs-list"
      role="tablist"
      style="outline: none;"
      tabindex="0"
    >
      <button
        aria-controls="radix-«r0»-content-all"
        aria-selected="true"
        class="data-[state=active]:bg-background dark:data-[state=active]:text-foreground focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:outline-ring dark:data-[state=active]:border-input dark:data-[state=active]:bg-input/30 text-foreground dark:text-muted-foreground inline-flex h-[calc(100%-1px)] flex-1 items-center justify-center gap-1.5 rounded-md border border-transparent px-2 py-1 text-sm font-medium whitespace-nowrap transition-[color,box-shadow] focus-visible:ring-[3px] focus-visible:outline-1 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:shadow-sm [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4"
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-slot="tabs-trigger"
        data-state="active"
        data-testid="tab-trigger-summary-all-heroes"
        id="radix-«r0»-trigger-all"
        role="tab"
        tabindex="-1"
        type="button"
      >
        All Characters (
        10
        )
      </button>
      <button
        aria-controls="radix-«r0»-content-favorites"
        aria-selected="false"
        class="data-[state=active]:bg-background dark:data-[state=active]:text-foreground focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:outline-ring dark:data-[state=active]:border-input dark:data-[state=active]:bg-input/30 text-foreground dark:text-muted-foreground h-[calc(100%-1px)] flex-1 justify-center rounded-md border border-transparent px-2 py-1 text-sm font-medium whitespace-nowrap transition-[color,box-shadow] focus-visible:ring-[3px] focus-visible:outline-1 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:shadow-sm [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4 flex items-center gap-2"
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-slot="tabs-trigger"
        data-state="inactive"
        data-testid="tab-trigger-summary-favorites-heroes"
        id="radix-«r0»-trigger-favorites"
        role="tab"
        tabindex="-1"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-heart h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2 9.5a5.5 5.5 0 0 1 9.591-3.676.56.56 0 0 0 .818 0A5.49 5.49 0 0 1 22 9.5c0 2.29-1.5 4-3 5.5l-5.492 5.313a2 2 0 0 1-3 .019L5 15c-1.5-1.5-3-3.2-3-5.5"
          />
        </svg>
        Favorites (
        1
        )
      </button>
      <button
        aria-controls="radix-«r0»-content-hero"
        aria-selected="false"
        class="data-[state=active]:bg-background dark:data-[state=active]:text-foreground focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:outline-ring dark:data-[state=active]:border-input dark:data-[state=active]:bg-input/30 text-foreground dark:text-muted-foreground inline-flex h-[calc(100%-1px)] flex-1 items-center justify-center gap-1.5 rounded-md border border-transparent px-2 py-1 text-sm font-medium whitespace-nowrap transition-[color,box-shadow] focus-visible:ring-[3px] focus-visible:outline-1 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:shadow-sm [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4"
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-slot="tabs-trigger"
        data-state="inactive"
        data-testid="tab-trigger-summary-only-heroes"
        id="radix-«r0»-trigger-hero"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Heroes (
        6
        )
      </button>
      <button
        aria-controls="radix-«r0»-content-villain"
        aria-selected="false"
        class="data-[state=active]:bg-background dark:data-[state=active]:text-foreground focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:outline-ring dark:data-[state=active]:border-input dark:data-[state=active]:bg-input/30 text-foreground dark:text-muted-foreground inline-flex h-[calc(100%-1px)] flex-1 items-center justify-center gap-1.5 rounded-md border border-transparent px-2 py-1 text-sm font-medium whitespace-nowrap transition-[color,box-shadow] focus-visible:ring-[3px] focus-visible:outline-1 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:shadow-sm [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4"
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-slot="tabs-trigger"
        data-state="inactive"
        data-testid="tab-trigger-summary-only-villain"
        id="radix-«r0»-trigger-villain"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Villains (
        4
        )
      </button>
    </div>
    <div
      aria-labelledby="radix-«r0»-trigger-all"
      class="flex-1 outline-none"
      data-orientation="horizontal"
      data-slot="tabs-content"
      data-state="active"
      data-testid="tab-content-summary-all-heroes"
      id="radix-«r0»-content-all"
      role="tabpanel"
      style="animation-duration: 0s;"
      tabindex="0"
    >
      <h1>
        Todos los personales
      </h1>
    </div>
    <div
      aria-labelledby="radix-«r0»-trigger-favorites"
      class="flex-1 outline-none"
      data-orientation="horizontal"
      data-slot="tabs-content"
      data-state="inactive"
      data-testid="tab-content-summary-favorites-heroes"
      hidden=""
      id="radix-«r0»-content-favorites"
      role="tabpanel"
      tabindex="0"
    />
    <div
      aria-labelledby="radix-«r0»-trigger-hero"
      class="flex-1 outline-none"
      data-orientation="horizontal"
      data-slot="tabs-content"
      data-state="inactive"
      data-testid="tab-content-summary-only-heroes"
      hidden=""
      id="radix-«r0»-content-hero"
      role="tabpanel"
      tabindex="0"
    />
    <div
      aria-labelledby="radix-«r0»-trigger-villain"
      class="flex-1 outline-none"
      data-orientation="horizontal"
      data-slot="tabs-content"
      data-state="inactive"
      data-testid="tab-content-summary-only-villain"
      hidden=""
      id="radix-«r0»-content-villain"
      role="tabpanel"
      tabindex="0"
    />
  </div>
  <div
    data-testid="mocked-hero-grid-section"
  >
    <h3>
      Mocked Hero Grid
    </h3>
    <div
      data-testid="card-123"
    >
      <h2>
        Hero 
        My Test hero
      </h2>
    </div>
  </div>
  <div
    data-testid="mocked-paginator-section"
  >
    <h3>
      Mocked CustomPagination section
    </h3>
    <h2>
      Total pages : 
      1
    </h2>
  </div>
</div>
`;

exports[`Testing HomePage.test > should show favorite tab 1`] = `
<div>
  <div
    data-testid="mocked-jumbotron-header"
  >
    Mocked CustomJumbotron
  </div>
  <div
    data-testid="mocked-herostats-section"
  >
    Mocked HeroStats section
  </div>
  <div
    class="flex flex-col gap-2 mb-8"
    data-orientation="horizontal"
    data-slot="tabs"
    dir="ltr"
  >
    <div
      aria-orientation="horizontal"
      class="bg-muted text-muted-foreground h-9 items-center justify-center rounded-lg p-[3px] grid w-full grid-cols-4"
      data-orientation="horizontal"
      data-slot="tabs-list"
      role="tablist"
      style="outline: none;"
      tabindex="0"
    >
      <button
        aria-controls="radix-«r5»-content-all"
        aria-selected="false"
        class="data-[state=active]:bg-background dark:data-[state=active]:text-foreground focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:outline-ring dark:data-[state=active]:border-input dark:data-[state=active]:bg-input/30 text-foreground dark:text-muted-foreground inline-flex h-[calc(100%-1px)] flex-1 items-center justify-center gap-1.5 rounded-md border border-transparent px-2 py-1 text-sm font-medium whitespace-nowrap transition-[color,box-shadow] focus-visible:ring-[3px] focus-visible:outline-1 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:shadow-sm [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4"
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-slot="tabs-trigger"
        data-state="inactive"
        data-testid="tab-trigger-summary-all-heroes"
        id="radix-«r5»-trigger-all"
        role="tab"
        tabindex="-1"
        type="button"
      >
        All Characters (
        10
        )
      </button>
      <button
        aria-controls="radix-«r5»-content-favorites"
        aria-selected="true"
        class="data-[state=active]:bg-background dark:data-[state=active]:text-foreground focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:outline-ring dark:data-[state=active]:border-input dark:data-[state=active]:bg-input/30 text-foreground dark:text-muted-foreground h-[calc(100%-1px)] flex-1 justify-center rounded-md border border-transparent px-2 py-1 text-sm font-medium whitespace-nowrap transition-[color,box-shadow] focus-visible:ring-[3px] focus-visible:outline-1 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:shadow-sm [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4 flex items-center gap-2"
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-slot="tabs-trigger"
        data-state="active"
        data-testid="tab-trigger-summary-favorites-heroes"
        id="radix-«r5»-trigger-favorites"
        role="tab"
        tabindex="-1"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-heart h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2 9.5a5.5 5.5 0 0 1 9.591-3.676.56.56 0 0 0 .818 0A5.49 5.49 0 0 1 22 9.5c0 2.29-1.5 4-3 5.5l-5.492 5.313a2 2 0 0 1-3 .019L5 15c-1.5-1.5-3-3.2-3-5.5"
          />
        </svg>
        Favorites (
        5
        )
      </button>
      <button
        aria-controls="radix-«r5»-content-hero"
        aria-selected="false"
        class="data-[state=active]:bg-background dark:data-[state=active]:text-foreground focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:outline-ring dark:data-[state=active]:border-input dark:data-[state=active]:bg-input/30 text-foreground dark:text-muted-foreground inline-flex h-[calc(100%-1px)] flex-1 items-center justify-center gap-1.5 rounded-md border border-transparent px-2 py-1 text-sm font-medium whitespace-nowrap transition-[color,box-shadow] focus-visible:ring-[3px] focus-visible:outline-1 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:shadow-sm [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4"
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-slot="tabs-trigger"
        data-state="inactive"
        data-testid="tab-trigger-summary-only-heroes"
        id="radix-«r5»-trigger-hero"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Heroes (
        6
        )
      </button>
      <button
        aria-controls="radix-«r5»-content-villain"
        aria-selected="false"
        class="data-[state=active]:bg-background dark:data-[state=active]:text-foreground focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:outline-ring dark:data-[state=active]:border-input dark:data-[state=active]:bg-input/30 text-foreground dark:text-muted-foreground inline-flex h-[calc(100%-1px)] flex-1 items-center justify-center gap-1.5 rounded-md border border-transparent px-2 py-1 text-sm font-medium whitespace-nowrap transition-[color,box-shadow] focus-visible:ring-[3px] focus-visible:outline-1 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:shadow-sm [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4"
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-slot="tabs-trigger"
        data-state="inactive"
        data-testid="tab-trigger-summary-only-villain"
        id="radix-«r5»-trigger-villain"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Villains (
        4
        )
      </button>
    </div>
    <div
      aria-labelledby="radix-«r5»-trigger-all"
      class="flex-1 outline-none"
      data-orientation="horizontal"
      data-slot="tabs-content"
      data-state="inactive"
      data-testid="tab-content-summary-all-heroes"
      hidden=""
      id="radix-«r5»-content-all"
      role="tabpanel"
      tabindex="0"
    />
    <div
      aria-labelledby="radix-«r5»-trigger-favorites"
      class="flex-1 outline-none"
      data-orientation="horizontal"
      data-slot="tabs-content"
      data-state="active"
      data-testid="tab-content-summary-favorites-heroes"
      id="radix-«r5»-content-favorites"
      role="tabpanel"
      style="animation-duration: 0s;"
      tabindex="0"
    >
      <h1>
        Favoritos
      </h1>
    </div>
    <div
      aria-labelledby="radix-«r5»-trigger-hero"
      class="flex-1 outline-none"
      data-orientation="horizontal"
      data-slot="tabs-content"
      data-state="inactive"
      data-testid="tab-content-summary-only-heroes"
      hidden=""
      id="radix-«r5»-content-hero"
      role="tabpanel"
      tabindex="0"
    />
    <div
      aria-labelledby="radix-«r5»-trigger-villain"
      class="flex-1 outline-none"
      data-orientation="horizontal"
      data-slot="tabs-content"
      data-state="inactive"
      data-testid="tab-content-summary-only-villain"
      hidden=""
      id="radix-«r5»-content-villain"
      role="tabpanel"
      tabindex="0"
    />
  </div>
  <div
    data-testid="mocked-hero-grid-section"
  >
    <h3>
      Mocked Hero Grid
    </h3>
    <div
      data-testid="card-456"
    >
      <h2>
        Hero 
        My Test hero 456
      </h2>
    </div>
    <div
      data-testid="card-678"
    >
      <h2>
        Hero 
        My Test hero 678
      </h2>
    </div>
  </div>
  <div
    data-testid="mocked-paginator-section"
  >
    <h3>
      Mocked CustomPagination section
    </h3>
    <h2>
      Total pages : 
      3
    </h2>
  </div>
</div>
`;
